lib/rouge/lexers/augeas.rb

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Augeas < RegexLexer
+      title "Augeas"
+      desc "The Augeas programming language (augeas.net)"
+
+      tag 'augeas'
+      aliases 'aug'
+      filenames '*.aug'
+      mimetypes 'text/x-augeas'
+
+      def self.reserved
+        @reserved ||= Set.new %w(
+          _ let del store value counter seq key label autoload
+          incl excl transform test get put in after set
+        )
+      end
+
+      def self.types
+        @types ||= Set.new %w(
+          unit string regexp lens tree filter
+        )
+      end
```

Comment:
Same as above.

I did have a question about `filter`. I noticed that in the visual sample, there's the line `let filter`. Does this mean that in this situation `filter` is being used as the name of a variable. The lexer lexes `filter` as a reserved keyword whenever it's used but perhaps that's wrong?

Reply:
Thanks for mentioning that. These keys are [types](https://augeas.net/docs/language.html) that can be passed for function variables. While it is true that `filter` is often used as a variable name for lens filters, it can also be used as a type.

Reply:
In that case, I think we can match types by simply having a rule that matching `:<type>`. That's what I've done in the latest version I've pushed. It means that words like `filter` won't highlight as types except when used to describe function parameters.